src/panels/lovelace/cards/hui-map-card.ts

```diff
@@ -325,11 +325,27 @@ class HuiMapCard extends LitElement implements LovelaceCard {
           continue;
         }
 
+        // create icon
+        let iconHTML = "";
+        const iconColor =
+          this._config!.dark_mode === true ? "#FFFFFF" : "#000000";
+        if (icon) {
+          const el = document.createElement("ha-icon");
+          el.setAttribute("icon", icon);
+          el.style.color = iconColor;
+          iconHTML = el.outerHTML;
+        } else {
+          const el = document.createElement("span");
+          el.innerHTML = title;
+          el.style.color = iconColor;
+          iconHTML = el.outerHTML;
+        }
+
         // create marker with the icon
         mapItems.push(
           Leaflet.marker([latitude, longitude], {
             icon: Leaflet.divIcon({
-              html: icon ? `<ha-icon icon="${icon}"></ha-icon>` : title,
+              html: iconHTML,
               iconSize: [24, 24],
               className: "",
```

Comment:
```
              html: icon ? `<ha-icon icon="${icon}"></ha-icon>` : title,
              iconSize: [24, 24],
              className: this._config!.dark_mode === true ? "dark" : "light",
            }),
```

Reply:
Add a class to the marker and define a class in `hui-map-card`: `.dark { color: #fafaf8;}`

Reply:
Implemented in https://github.com/home-assistant/home-assistant-polymer/pull/4085/commits/9d10ab8f054f4dcf50bc10b0768c8194b4359c2b